pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -713,7 +724,7 @@ ${compileResult.stdout}
 
     if (input is BuildInput) {
       final packagesWithLink =
-          (await packageLayout.packagesWithAssets(Hook.link))
+          (await (await _planner).packagesWithHook(Hook.link))
```

Comment:
Nested `await`s do make my skin crawl a bit ;)

Reply:
Perhaps
```
final planner = await _planner;
final packagesWithHook = await planner.packagesWithHook(Hook.build);
```
further above could be extracted to a method to alleviate this here and in other places, as the `_planner` is mostly accessed to run `packagesWithHook`. Just a nit.

Reply:
Agreed with the skin crawl, I already adopted `final planner = await _planner;` in other places.